BigImageViewer/src/main/java/com/github/piasy/biv/view/ImageViewFactory.java

```diff
@@ -37,29 +39,51 @@
  */
 public class ImageViewFactory {
 
-    public final View createMainView(Context context, int imageType, File imageFile,
-            int initScaleType) {
+    public final View createMainView(final Context context, final int imageType, final int initScaleType) {
         switch (imageType) {
             case ImageInfoExtractor.TYPE_GIF:
             case ImageInfoExtractor.TYPE_ANIMATED_WEBP:
-                return createAnimatedImageView(context, imageType, imageFile, initScaleType);
+                return createAnimatedImageView(context, imageType, initScaleType);
             case ImageInfoExtractor.TYPE_STILL_WEBP:
             case ImageInfoExtractor.TYPE_STILL_IMAGE:
             default:
                 return createStillImageView(context);
         }
     }
 
-    protected SubsamplingScaleImageView createStillImageView(Context context) {
+    public boolean isAnimatedContent(final int imageType) {
+        switch (imageType) {
+
+            case ImageInfoExtractor.TYPE_GIF:
+            case ImageInfoExtractor.TYPE_ANIMATED_WEBP:
+                return true;
+
+            default:
+                return false;
+        }
+    }
+
+    protected SubsamplingScaleImageView createStillImageView(final Context context) {
         return new SubsamplingScaleImageView(context);
     }
 
-    protected View createAnimatedImageView(Context context, int imageType, File imageFile,
-            int initScaleType) {
+    protected View createAnimatedImageView(final Context context, final int imageType, final int initScaleType) {
         return null;
     }
 
-    public View createThumbnailView(Context context, Uri thumbnail, ImageView.ScaleType scaleType) {
-        return null;
+    public void loadAnimatedContent(final View view, final int imageType, final File imageFile) {}
+
+    public void loadSillContent(final View view, final Uri uri) {
+        if (view instanceof SubsamplingScaleImageView) {
+            ((SubsamplingScaleImageView) view).setImage(ImageSource.uri(uri));
+        }
+    }
+
+    public View createThumbnailView(final Context context, final ImageView.ScaleType scaleType) {
+        return new ImageView(context);
+    }
+
+    public void loadThumbnailContent(final View view, final Uri thumbnail) {
+        ((ImageView) view).setImageURI(thumbnail);
```

Comment:
Please check if it's an `ImageView` too.

Reply:
done!